modules/javafx.graphics/src/main/java/javafx/animation/Animation.java

```diff
@@ -978,12 +979,15 @@ public void playFromStart() {
      * Note: <ul>
      * <li>{@code play()} is an asynchronous call, the {@code Animation} may not
      * start immediately. </ul>
+     * <p>
+     * This method must be called on the JavaFX Application thread.
      *
-     * @throws IllegalStateException
-     *             if embedded in another animation,
+     * @throws IllegalStateException if this method is called on a thread
+     *                other than the JavaFX Application Thread, or if embedded in another animation,
      *                such as {@link SequentialTransition} or {@link ParallelTransition}
      */
     public void play() {
+        Toolkit.getToolkit().checkFxUserThread();
```

Comment:
minor: perhaps this should first check for non-null parent, then for fx thread (here and below)

Reply:
Our pattern everywhere else where we require the method to be called on the FX Application Thread is to do the thread check first.

Reply:
I don't have a preference, but I see that this check is usually the first one being performed, and then others (see for instance `QuantumToolkit::enterNestedEventLoop`.